CalenderInterface Implementation

Shuviu opened the pull request:
- Implemented calender_wrapper.py as abstract base class for CalenderImplementations

- Implemented calender_event.py as dataclass for storing informations about fetched calender events / appointments

Shuviu committed:
CalenderInterface Implementation

- Implemented calender_wrapper.py as abstract base class for CalenderImplementations

- Implemented calender_event.py as dataclass for storing informations about fetched calender events / appointments

Shuviu committed:
CalenderInterface Implementation

- Implemented calender_wrapper.py as abstract base class for CalenderImplementations

- Implemented calender_event.py as dataclass for storing informations about fetched calender events / appointments

Shuviu committed:
Updated calender_wrapper.py

- Updatad proterties to be non abstract

- Updated fetch_events to fetch_events_from_range to add time frame functionality

- Added fetch_events_from today

Shuviu committed:
Updated calender_wrapper.py

- Updatad proterties to be non abstract

- Updated fetch_events to fetch_events_from_range to add time frame functionality

- Added fetch_events_from today

theSys7em commented on src/remindhub_device/calender_wrapper.py:
```diff
@@ -0,0 +1,35 @@
+        pass
+
+    def fetch_events_from_today(self) -> List[CalenderEvent]:
+        return self.fetch_events_from_range(datetime.datetime.today(), datetime.datetime.today())
```
Uses identical datetimes
eg from 08.06.2024 13:43 till 08.06.2024 13:43

theSys7em commented on src/remindhub_device/calender_wrapper.py:
```diff
@@ -0,0 +1,35 @@
+
+    @property
+    def calender_url(self) -> str:
+        return self.calender_url
```
The way you implemented properties results in an infinite loop since it is calling the method itself again and not a property you set

theSys7em commented on src/remindhub_device/calender_wrapper.py:
```diff
@@ -0,0 +1,35 @@
+from src.remindhub_device.calender_event import CalenderEvent
+
+
+class CalenderWrapper(ABC):
```
Since the implementations will inherit from this class the name CalenderWrapper is not good practice
You should just call it CalendarProvider

theSys7em commented on src/remindhub_device/calender_event.py:
```diff
@@ -0,0 +1,10 @@
+import datetime
+
+
```
Use the @dataclass Annotation for dataclasses
This enabled you to delete the constructor and don't worry about instance management

theSys7em reviewed (changes requested):
See comments

Shuviu committed:
Updating calender_provider.py

- renamed calender_wrapper.py to calender_provider.py to

- Updatad proterties

- Updated fetch_events_from_today do use current and +1 date

- Added Init method

- Added @DataClass annotation to calender_event.py

(Sorry first time python user D:)

Shuviu committed:
Updating calender_provider.py

- renamed calender_wrapper.py to calender_provider.py to

- Updatad proterties

- Updated fetch_events_from_today do use current and +1 date

- Added Init method

- Added @DataClass annotation to calender_event.py

(Sorry first time python user D:)

theSys7em committed:
Correct typos in class and file names: renamed "Calender" to "Calendar" and updated imports and references accordingly.

Renamed "CalendarEvent" to "CalendarEntry"
Optimized imports and dataclass CalendarEntry

theSys7em committed:
Correct typos in class and file names: renamed "Calender" to "Calendar" and updated imports and references accordingly.

Renamed "CalendarEvent" to "CalendarEntry"
Optimized imports and dataclass CalendarEntry

theSys7em commented on src/remindhub_device/calendar_provider.py:
```diff
@@ -0,0 +1,44 @@
+        pass
+
+    def fetch_events_from_today(self) -> List[CalendarEntry]:
+        return self.fetch_events_from_range(datetime.datetime.today(), datetime.datetime.today() + datetime.timedelta(days=1))
```
Calendar should return entries of today, even the ones that might be in the past already
So from today midnight till tomorrow midnight
Calendar Entries that are already in the past will be displayed differently in the UI but should still be visible

theSys7em committed:
Merge remote-tracking branch 'origin/1-calenderinterface' into 1-calenderinterface

Shuviu committed:
Updating calender_provider.py

- Updated fetch_events_from_today to use the correct times

Shuviu committed:
Merge remote-tracking branch 'origin/1-calenderinterface' into 1-calenderinterface

theSys7em reviewed (approved):
Looks good
